crates/proto/src/op/message.rs

```diff
@@ -657,7 +657,7 @@ impl Message {
     ) -> ProtoResult<(Vec<Record>, Option<Edns>, Vec<Record>)> {
         let mut records: Vec<Record> = Vec::with_capacity(count);
         let mut edns: Option<Edns> = None;
-        let mut sigs: Vec<Record> = Vec::with_capacity(if is_additional { 1 } else { 0 });
+        let mut sigs: Vec<Record> = Vec::with_capacity(usize::from(is_additional));
```

Comment:
Similar comment, I'm not sure this change makes this more obvious. In fact, I might disable this warning globally...